docs/go-configuration.md

```diff
@@ -4,27 +4,366 @@ title: Scope Go Agent configuration
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Environment Variable-->
 
 ```sh
 SCOPE_TESTING_MODE=false
 ```
+
+<!--By Code-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithTestingModeEnabled()))
+}
+
+// ...
+```
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+## Enabling test retries on failure
+
+The Scope agent has support for execution retries on failing tests. By default, no retries are configured.
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Environment Variable-->
+
+```sh
+SCOPE_TESTING_FAIL_RETRIES=3
+```
+
+<!--By Code-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithRetriesOnFail(3)))
+}
+
+// ...
+```
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+## Handling test panic as a failure
+
+By default, the standard testing framework exits the test process on panic. This behavior can be changed to handle panics as fails, allowing the complete execution of the test suite. 
+
+Also, this option can be combined with test retries, to retry tests that panic.
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Environment Variable-->
+
+```sh
+SCOPE_TESTING_PANIC_AS_FAIL=true
+```
+
+<!--By Code-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithHandlePanicAsFail()))
+}
+
+// ...
+```
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+## Adding agent metadata
+
+You can send arbitrary metadata for every test associated with certain commit which will be shown by Scope.
+
+Additionally, it is possible to set an environment variable as metadata value for a certain key, whose final value will be evaluated at runtime.
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Environment Variable-->
+
+```sh
+SCOPE_METADATA="CustomKey1=MyValue, CustomKey2=MyValue2, CustomPath=\$PATH"
+```
+
+This is equivalent to: 
+```go
+    map[string]interface{}{
+		"CustomKey1": "MyValue",
+		"CustomKey2": "MyValue2",
+		"CustomPath": os.Getenv("PATH"),
+	}
+```
+
+<!--By Code-->
+
+```go
+import (
+    "os"
+    "testing"
+
+    "go.undefinedlabs.com/scopeagent"
+    "go.undefinedlabs.com/scopeagent/agent"
+)
+
+func TestMain(m *testing.M) {
+    os.Exit(scopeagent.Run(m, agent.WithMetadata(map[string]interface{}{
+		"CustomKey1": "MyValue",
+		"CustomKey2": "MyValue2",
+		"CustomPath": os.Getenv("PATH"),
+	})))
+}
+
+// ...
+```
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+## Select agent metadata as test configuration
+
+Yon can select metadata keys to be considered as relevant in the configuration of the test.
```

Comment:
For the future, we should have a section that explains what a _configuration_ is.

Reply:
ok